src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -44,21 +49,49 @@ const EditShiftHours = ({
 
   const inputName = 'shift';
   const { setServiceError, userId } = useApplicationContext();
+  const [isChecked, setIsChecked] = useState(false);
+  const [redirectTarget, setRedirectTarget] = useState('');
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const [clashingProperty, setClashingProperty] = useState(null);
+  const [clashingTimes, setClashingTimes] = useState(null);
+  const navigate = useNavigate();
+
   const {
     timeEntries,
     setTimeEntries,
     timecardDate,
     summaryErrors,
     setSummaryErrors,
+    setIsAlertVisible,
+    summaryMessages,
+    setSummaryMessages,
   } = useTimecardContext();
 
-  const [clashingProperty, setClashingProperty] = useState(null);
-  const [clashingTimes, setClashingTimes] = useState(null);
+  const startEntryExists = !!timeEntry?.startTime && timeEntry.startTime !== '';
+  const finishEntryExists =
+    !!timeEntry?.finishTime && timeEntry.finishTime !== '';
+  const [localStartDate, setLocalStartDate] = useState(
+    startEntryExists ? timeEntry.startTime : timecardDate
+  );
+  const [localEndDate, setLocalEndDate] = useState(
+    finishEntryExists ? timeEntry.finishTime : timecardDate
+  );
 
   useEffect(() => {
     focusErrors(document.getElementById('summary-error-0-message'));
+    if (shouldRedirect) {
+      navigate(redirectTarget);
+    }
+
+    if (summaryMessages && Object.keys(summaryMessages).length !== 0) {
+      setTimeout(() => setIsAlertVisible(false), 10000);
+    }
```

Comment:
Just thinking that these get called when `summaryErrors` changes, which seems to be from line 240 when they are reset. It might be better to move this logic into another useEffect which gets called when `summaryMessages` changes

Reply:
Or a function that gets called after `setSummaryMessages` on line 176?

Reply:
Created second useEffect and split the code out